source/develop/release-management/features/network/import-vm-net-changes-2.html.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Expand vNic profile mapping options for VM import from Storage Domain via REST API
+category: feature
+authors: eraviv
+feature_name: Expand vNic profile mapping options for VM import from Storage Domain via REST API
+feature_modules: engine backend/api-model
+feature_status: release 4.2
+---
+
+# Expand vNic profile mapping options for VM import from Storage Domain via REST API
+
+## Summary
+
+Up to oVirt Engine 4.2, importing a VM from a storage domain via the REST API 
+assumed full knowledge of the target engine profile IDs and employed a fail-fast 
+failure policy:
+
+* Mapping a vNic profile by profile ID (see import-vm-net-changes) requires the
+  user to discover the profile IDs on the engine in advance and to keep them 
+  updated.
+* On any error or mismatch in the mapping of the target or the source of any vNic, 
+  the request is aborted.
+
+Requiring a target ID for the import is labor-intensive from the perspective of
+the user.
+Employing a fail-fast failure policy is desirable when the user would like the 
+engine to guard against incomplete or faulty inputs. But when the import is part 
+of a Disaster Recovery operation, a permissive behaviour is required in order to 
+enable the user to complete the import even if some of the vNics would not be 
+fully mapped. 
+
+This feature intends to solve the above issues for importing VMs from a 
+data-domain storage via the REST API of the engine.
+
+## Owner
+
+*   Name: Eitan Raviv
+*   email: <eraviv@redhat.com>
+
+## Solution
+
+### General behaviour
+
+Additional functionality has been added to the engine:
+
+* Deprecate the old restrictive mode, and add a new permissive mode.
+* The permissive mode supports specifying the target profile either using a
+  profile id or using a pair of profile name and network name. This pair is
+  unique at the data-center level.
+* The permissive mode implements an escalating matching algorithm that tries
+   to find the best match for the profile on the engine: 
+	* try to match by target id specified in the user mapping
+	* else, try to match by target profile name + network name
+	* else, try to match by source profile name + network name
+	  found on the vNic in the OVF
+* If none of the above succeeds, the permissive mode applies a 'no profile' 
+  target to the vNic and continues with the request.
+* User input for vNic mappings joins all the other inputs of the import under
+  the 'registration_configuration' entry in the request body. 
+
```

Comment:
Could we have a brief description of what happens to the nic if 'no profile' is applied.
Also a brief not of how to fix the nic after nic import (I assume it's just going to the UI and choosing a profile)

Reply:
@maorlipchuk can you please review?

Reply:
@erav can you please address @mmirecki comment?

Reply:
@mburman5 can you please review?

Reply:
added comment per @mmirecki comment

Reply:
If no profile is applied is should be 'Empty' by default. Eitan, is this is the case?

Reply:
@mburman5 - this was true in the past but it it no longer the case - the requirement has now changed. To explicitly signal that an 'Empty' profile is desired, usage of empty strings in the target network name and profile name should be employed. If no target is specified at all, then the cascading algorithm described here will be employed, which in this case means that the source network\profile will be the next best candidates for the vNic.